pypsa/opf.py

```diff
@@ -1647,6 +1649,8 @@ def network_lopf(network, snapshots=None, solver_name="glpk", solver_io=None,
     -------
     None
     """
+    if multi_investment_periods or (type(network.snapshots)==pd.MultiIndex):
+                raise NotImplementedError("Multi period invesmtent is only supported for pyomo=False")
```

Comment:
Mhm, we can think about writing an extra function converting a pypsa network with multiindex to single-index. The pandas function to_flat_index() does not work, because it converts the multiindex into a single index with tuples, which in turn results in an error in the lopf(). Also setting the snapshots from multiindex to single index results currently in an error since the snapshot_weightings (with multiindex) are tried to be reindexed with single index [here](https://github.com/PyPSA/PyPSA/blob/601098eaa132868587067501487670a04e487eb4/pypsa/components.py#L396-L400).

I am still favoring the way it was implemented in the [multi-decade branch](https://github.com/PyPSA/PyPSA/blob/f4311fb6d8a4a5f8ad1ca782f2c3ca5ae0edd321/pypsa/components.py#L376) because reindexing was possible in all 4 directions (network snapshots multi or single index, new snapshots multi or single index) and the time-varying data was kept. Maybe we can add an extra switch in the function set_snapshots(time_varying=True) to keep the time-varying data?

Reply:
I'd keep it simple and say multi investment period optimization and optimization with multi-indexed snapshots is not supported with pyomo=True